speechbrain/core.py

```diff
@@ -510,89 +177,28 @@ class and override any methods for which the default behavior does not
 
     Arguments
     ---------
-    modules : dict of str:torch.nn.Module pairs
+    modules : dict[str, torch.nn.Module]
         These modules are passed to the optimizer by default if they have
         trainable parameters, and will have ``train()``/``eval()`` called on them.
-    opt_class : torch.optim class
+    opt_class : Optional[Type[torch.optim]]
         A torch optimizer constructor that takes only the list of
         parameters (e.g. a lambda or partial function definition). By default,
         this will be passed all modules in ``modules`` at the
         beginning of the ``fit()`` method. This behavior can be changed
         by overriding the ``configure_optimizers()`` method.
-    hparams : dict
+    hparams : Optional[dict]
```

Comment:
I am wondering if we should maybe log this to the user i.e. saying that no opt+class / hparams are provided etc to make sure everything is transparent to the user